### ISSUE TITLE
Rough ideas on building environment

### ISSUE THREAD

**Issue by Yajieniu** [PatoBeltran/RLMap#1]
I like your idea of collaborating. I have some rough ideas (because I haven't tested them), thanks to TA Andrew. 

**Preparation:** 
Represent car/pedestrian/bike/traffic light/road/etc as classes in python. Each of them has certain characters, like for a car: on which lane it's running, the speed, acceleration, direction, position (we can use numbers to show the position on a lane). 

class Car(object):
	color = ""
	lane = 0
	speed = 0
	acceleration = 0
	direction = 0
        position = ""

	def __init__(self, color, lane, speed, acceleration, direction, position)
		self.color = color
		self.lane = lane
		self.speed = speed
		self.acceleration = acceleration
		self.direction = direction
                self.position = position

def make_car(color, lane, speed, acceleration, direction)
	car = Car(color, lane, speed, acceleration, direction)
	return car

And a pedestrian class can have speed (walking speed), direction (from lane A to lane B), position, for example, the following 'picture'
               _____________________________________________________
lane 1:     ________________________________________________
lane 2:    ________________________________________________
lane 3:     ____________________________________ car 2 ____________
lane 4:     __________ car 1 ____________ Y __________________________

Y is a pedestrian starting from that position and trying to cross the road from lane 4 to lane 1.

Similarly, we can build classes for bike/road/traffic light, etc.  For road, we can decide how long it is. A car passed the end of a lane should show up again at the beginning of that lane.

**Environment --  a program**
We could write a program to execute the functions of the environment. It can:
1. automatically create new cars, new pedestrians, new bikes, and so on; and assign them speed, direction, etc
2. keep track of the position of each car, bike, pedestrian, etc. If any of the cars has the same position with another car or pedestrian --- lose --- give bad reward 
3. keep track of the signal of traffic light(s), if any car runs a red light -- lose -- give bad reward
4. to be continued.
....

Basically, the environment should be a working world, be anything other than our car. Once we build it, it should have the ability to automatically run from start to the end, and give a reward.

If you find this helpful, I would love to discuss more with you. I will start working on it today.

My github username is Yajieniu. Please add me to the repo. Thanks.

**Comment by vkhetan** [comment]
Guys, I am also looking forward to collaborate, have you heard of Tinker library in python

**Comment by Yajieniu** [comment]
No, I never used it. How can it help?

**Comment by victoriasovereigne** [comment]
I used Tkinter for my world, but I don't know how I can build RL from it. Currently there are only 4 dumb cars and no traffic lights yet.

![image](https://cloud.githubusercontent.com/assets/7977053/24775540/f11fb378-1ae2-11e7-9501-6afc9ffde843.png)

**Comment by vkhetan** [comment]
It looks great, add me to this repo and can we meet today afternoon.

**Comment by brennandubuc** [comment]
Thanks for all that info Yajie

I wasn't aware that we need to associate rewards with the other cars in our world. I thought that only our car would be controlled by rewards; the other cars just acting like moving obstacles. Is that correct?

**Comment by PatoBeltran** [comment]
Sorry guys I was working on other projects. I added you to the project as collaborators, what do you think would be the best approach to start with all of this?

**Comment by vkhetan** [comment]
Hey, thanks, man. I saw it but I did a lot of work and don't want to redo it now.

**Comment by PatoBeltran** [comment]
@vkhetan you could merge your work with this repo and we can all help you get it finished/improve
